universal-application-tool-0.0.1/app/services/question/QuestionService.java

```diff
@@ -0,0 +1,76 @@
+package services.question;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+
+public interface QuestionService {
+
+  /**
+   * Creates a new Question Definition. Returns a QuestionDefinition object on success and null on
+   * failure.
+   *
+   * <p>This will fail if he path provided already resolves to a QuestionDefinition or Scalar.
+   *
+   * <p>NOTE: This does not update the version.
+   */
+  Optional<QuestionDefinition> create(QuestionDefinition definition);
```

Comment:
Thought (non-blocking): This is making me wonder where the right place to detect and surface validation errors is. This *could* return an `ErrorsOr<QuestionDefinition>` where the outer class is something like

```java
class ErrorsOr<T> {
  private final Optional<T> wrapped;
  private final ImmutableMap<String, String> errors;

  ErrorsOr(T wrapped) {
    this.wrapped = Optional.of(wrapped);
    this.errors = ImmutableMap.of();
  }

  ErrorsOr(ImmutableMap<String, String> errors) {
    this.wrapped = Optional.empty();
    this.errors = errors;
  }
  
  // getters and predicate methods
}
```

Or we could do validation at the form-object level.